tcp/pcb.ml

```diff
@@ -465,6 +468,10 @@ struct
       Lwt.return_unit
     | None ->
       Tx.send_rst t id ~sequence ~ack_number ~syn ~fin
+    | exception ex ->
+      PCB_Log.err (fun f -> f "uncaught exception looking up handler for TCP port %d:@ %s"
+                      id.WIRE.local_port (Printexc.to_string ex));
+      Lwt.return_unit
```

Comment:
well, we can decide to be completely defensive in all external APIs, but imho this would make the code a mess (and likely does checking more than once).  I'd rather strive for 'no exceptions at module boundaries', and thus if a client provides some function which fails, we fail hard.  if there are potential errors, we should use a `result` type. in this specific case I think an `option` is a good solution.

what should be handled defensively is input from the network.